x-pack/metricbeat/module/gcp/distribution.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package gcp
+
+import (
+	"math"
+
+	"google.golang.org/genproto/googleapis/api/distribution"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+func containsHistogram(d *distribution.Distribution) bool {
+	if d.Count == 0 || d.BucketOptions == nil || len(d.BucketCounts) == 0 {
+		return false
+	}
+
+	var sum int64
+
+	for _, v := range d.BucketCounts {
+		sum += v
+	}
+
+	// Count must be equal to the sum of values in BucketCounts.
+	return d.Count == sum
+}
+
+func calcExplicitUpperBound(bucket *distribution.Distribution_BucketOptions_Explicit, i int) float64 {
+	return bucket.Bounds[i]
+}
+
+func calcExponentialUpperBound(bucket *distribution.Distribution_BucketOptions_Exponential, i int) float64 {
```

Comment:
Is there any documentation we can add or comments in the code to explain the different histograms (explicit, exponential, linear...) and what are the bucket options available from GCP?